rust/runtime/src/module/dylib.rs

```diff
@@ -0,0 +1,90 @@
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    os::raw::{c_int, c_void},
+};
+
+use tvm_common::{ffi::BackendPackedCFunc, packed_func::PackedFunc};
+
+use crate::{
+    threading::{TVMBackendParallelBarrier, TVMBackendParallelLaunch},
+    workspace::{TVMBackendAllocWorkspace, TVMBackendFreeWorkspace},
+    TVMAPISetLastError,
+};
+
+use super::Module;
+
+pub struct DylibModule<'a> {
+    lib: libloading::Library,
+    packed_funcs: RefCell<HashMap<String, &'a (dyn PackedFunc)>>,
+}
+
+macro_rules! init_context_func {
+    ($lib:ident, $( ($fn:ident, $sig:ty) ),+ $(,)?) => {
+        unsafe {
+            $(
+                let fn_ptr = $lib.get::<*mut $sig>(concat!("__", stringify!($fn)).as_bytes());
+                if let Ok(fn_ptr) = fn_ptr {
+                    **fn_ptr = $fn;
```

Comment:
Check if you would like to handle these cases https://github.com/dmlc/tvm/blob/master/src/runtime/dso_module.cc#L71

- The ctx_addr is used to handle callback when the library tries to callback to a PackedFunc that is local to the current module
- The dev_mblob records other modules that this module imported through a binary blob